src/main/groovy/org/shipkit/internal/util/RepositoryNameUtil.java

```diff
@@ -0,0 +1,49 @@
+package org.shipkit.internal.util;
+
+import org.shipkit.internal.gradle.util.StringUtil;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RepositoryNameUtil {
```

Comment:
Nice utility class that encapsulates complexity in a single spot, making the codebase easy to read. Cool :)